src/parser/priest/discipline/modules/features/AtonementApplicatorBreakdown.js

```diff
@@ -0,0 +1,256 @@
+        this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.ATONEMENT_HEAL_NON_CRIT), this.handleAtonementsHits);
+        this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.ATONEMENT_HEAL_CRIT), this.handleAtonementsHits);
+    }
+
+    storeShadowMendsCasts(event){
+        this._castsAplyBuffsMap.set({
+            "event": event,
+            "applicatorId": SPELLS.SHADOW_MEND.id,
+        }, null);
+    }
+    
+    storePowerWordShieldsCasts(event){
+        this._castsAplyBuffsMap.set({
+            "event": event,
+            "applicatorId": SPELLS.POWER_WORD_SHIELD.id,
+        }, null);
+
+    }
+
+    assignAtonementBuffToApplicator(event){
+        if(event.__modified === true){ // Power Word: Radiance
+
+            //Set the wasRefreshed property of the old atonement on the same target to true
+            //so we can stop attributing atonement healing to the old atonement
+            if(event.type === "refreshbuff"){
+                this.setWasRefreshedProperty(event, true);
+            }
+
+            //Putting a custom event object for Radiances since there it's only 1 cast for 5 buffs
+            this._castsAplyBuffsMap.set({
+                "event": {
+                    "targetID": event.targetID,
+                },
+                "applicatorId": SPELLS.POWER_WORD_RADIANCE.id,
+            }, 
+            {
+                "applyBuff": event,
+                "atonementEvents": [],
+                "extendedByEvangelism": false,
+                "wasRefreshed": false,
+            });
+        } else { //Shadow Mend and Power Word: Shield
+            if(event.type === "refreshbuff"){
+                this.setWasRefreshedProperty(event, true);
+            }
+
+            //Iterate the casts in reverse to break out at the most recent appearance of the corresponding targetID
+            const playerWithAtonement = event.targetID;
+            const reversedMapKeys = Array.from(this._castsAplyBuffsMap.keys()).slice().reverse();
+
+            for(const cast of reversedMapKeys) {
+                if(cast.event !== null){
+                    if(cast.event.targetID === playerWithAtonement){
+                        this._castsAplyBuffsMap.set(cast, {
+                            "applyBuff": event,
+                            "atonementEvents": [],
+                            "extendedByEvangelism": false,
+                            "wasRefreshed": false,
+                        });
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    handleAtonementsHits(event){
+        //Healing from atonements pre-applied before entering combat
+        //will assume PW:S as the applicator since it's usually the most common one used pre-pull,
+        const values = this._castsAplyBuffsMap.values();
+        if(Array.from(values).find(atonement => atonement.applyBuff.targetID === event.targetID) === undefined){
```

Comment:
This could benefit from a named temp var. A well named var could clarify your intent.